openff/toolkit/utils/utils.py

```diff
@@ -207,12 +215,11 @@ def string_to_quantity(quantity_string) -> Union[str, int, float, Quantity]:
 
     # TODO: Should intentionally unitless array-likes be Quantity objects
     #       or their raw representation?
-    if (quantity.units == unit.dimensionless) and isinstance(quantity.m, (int, float)):
+    if str(quantity.units) == "dimensionless" and isinstance(quantity.m, (int, float)):
```

Comment:
Only relevant since it's a high-trafficked line

```
In [1]: from openff.units import unit, Quantity, Unit

In [2]: _DIMENSIONLESS = Unit("dimensionless")

In [3]:

In [3]: x = Quantity("1.0")

In [4]: %%timeit
   ...: x.units == unit.dimensionless
   ...:
   ...:
4.93 μs ± 69.1 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [5]: %%timeit
   ...: str(x.units) == "dimensionless"
   ...:
   ...:
2.5 μs ± 90.4 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [6]: %%timeit
   ...: x.units == _DIMENSIONLESS
   ...:
   ...:
975 ns ± 56.1 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```